lib/queuePopulator/LogReader.js

```diff
@@ -379,6 +390,46 @@ class LogReader {
         return undefined;
     }
 
+    _processFilterEntry(batchState, record, done) {
+        const { entriesToPublish } = batchState;
+        if (!record || !record.entries) {
+            return done();
+        }
+        this._setEntryBatch(entriesToPublish);
+        return this._filterEntries(batchState, record, err => {
+            this._unsetEntryBatch();
+            if (err) {
+                this.log.error('error filtering entries from log', {
+                    method: 'LogReader._processFilterEntries',
+                    error: err,
+                });
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    _processFilterEntries(batchState, done) {
+        const { currentRecords } = batchState;
+        if (!currentRecords || !(currentRecords.length > 0)) {
```

Comment:
Could also be simply checked with `currentRecords.length === 0`, I will not block the PR for this though :wink: